server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -635,11 +635,38 @@ public CommitStats commitStats() {
     }
 
     /**
-     * The sequence number service for this engine.
+     * @return the local checkpoint for this Engine
+     */
+    public abstract long getLocalCheckpoint();
+
+    /**
+     * @return the maximum sequence number from the local checkpoint tracker
+     */
+    public abstract long getMaxSeqNo();
+
+    /**
+     * @return generate a new sequence number from the local checkpoint tracker
+     */
+    public abstract long generateSeqNo();
```

Comment:
this is super dangerous IMO. Maybe we can make `getLocalCheckpointTracker` package private and have a testing utility to get it? I think that's good enough for tests and we can keep it isolated from production code?

Reply:
Yep, I've removed this